sdk/resourcemanager/azure-resourcemanager-containerinstance/src/main/java/com/azure/resourcemanager/containerinstance/models/ContainerGroup.java

```diff
@@ -922,6 +922,78 @@ WithContainerInstanceAttach<ParentT> withReadOnlyVolumeMountSetting(
                     Map<String, String> volumeMountSetting);
             }
 
+            /**
+             * The stage of the container instance definition allowing to specify liveness probe.
+             * Azure Container Instances supports liveness probes so that you can configure your containers within your
+             * container group to restart if critical functionality is not working.
+             *
+             * @param <ParentT> the stage of the parent definition to return to after attaching this definition
+             */
+            interface WithLivenessProbe<ParentT> {
+                /**
+                 * Specifies the container's liveness probe to execute a given command at a given interval.
+                 *
+                 * @param command the command for the probe to execute
+                 * @param probePeriodSeconds the interval at which the command executes
+                 * @return the next stage of the definition
+                 */
+                WithContainerInstanceAttach<ParentT> withLivenessProbeExec(List<String> command, int probePeriodSeconds);
```

Comment:
I am not sure if `Exec` is the best phrase here. Maybe full `ExecutionCommand`, or `Command`, or just `Execution`?

Reply:
Should we note that only 1 liveness probe can be set (if it is the case)?

Reply:
Then probably `ExecutionCommand` or `Command`. May ref their doc https://learn.microsoft.com/en-us/azure/container-instances/container-instances-liveness-probe#liveness-command

Reply:
Changed. Also added overload for failureThreshold.